### PR TITLE
[cleanup] remove deletedTreeCleanUpLevel feature flag

### PR DESCRIPTION
**Pull request by kassens**
I noticed this was an experiment concluded 16 months ago (#21679) that this extra work is beneficial
to break up cycles leaking memory in product code.

### CODE REVIEW

File: packages/react-reconciler/src/ReactFiberCommitWork.new.js [hunk 1]
```diff
@@ -1661,74 +1660,43 @@ function detachFiberAfterEffects(fiber: Fiber) {
...
-    fiber.sibling = null;
-    fiber.stateNode = null;
-    fiber.updateQueue = null;
-
-    if (__DEV__) {
-      fiber._debugOwner = null;
+  // Clear cyclical Fiber fields. This level alone is designed to roughly
+  // approximate the planned Fiber refactor. In that world, `setState` will be
+  // bound to a special "instance" object instead of a Fiber. The Instance
+  // object will not have any of these fields. It will only be connected to
```
Comment (kassens) [inline packages/react-reconciler/src/ReactFiberCommitWork.new.js:1666]: I think Fiber has been completely rolled out? It wasn't quite clear to me how to update this comment as the fields still exist with Fiber.

> The Instance object will not have any of these fields

File: packages/react-reconciler/src/ReactFiberCommitWork.js [hunk 2]
```diff
@@ -4196,8 +4162,7 @@ function commitPassiveUnmountEffectsInsideOfDeletedTree_begin(
     resetCurrentDebugFiberInDEV();
 
     const child = fiber.child;
-    // TODO: Only traverse subtree if it has a PassiveStatic flag. (But, if we
-    // do this, still need to handle `deletedTreeCleanUpLevel` correctly.)
```
Comment (alunyov) [inline packages/react-reconciler/src/ReactFiberCommitWork.js:4200]: What does this mean? 

> But, if we do this, still need to handle `deletedTreeCleanUpLevel` correctly.

Comment (kassens) [reply packages/react-reconciler/src/ReactFiberCommitWork.js:4200]: Not quite sure, but that flag no longer exists, so I removed this part of the comment…

**Review by sammy-SC — APPROVED** [review]

File: packages/react-reconciler/src/ReactFiberCommitWork.new.js (hunk 1 above)
Comment (sennymunoz) [reply packages/react-reconciler/src/ReactFiberCommitWork.new.js]: Oh yeah

**Pull request merged**